From 5dbe8ec3aebc0c282b41e45bdc93c9c00a2368d3 Mon Sep 17 00:00:00 2001
From: Alexey Dobriyan <adobriyan@openvz.org>
Date: Mon, 17 Mar 2008 14:38:48 +0300
Subject: [PATCH 33/48] NETFILTER: fix ipv4 conntrack structure double free

It's masked by it's NULLifying, but it's double free nonetheless.
---
 net/ipv4/netfilter/nf_conntrack_l3proto_ipv4.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/net/ipv4/netfilter/nf_conntrack_l3proto_ipv4.c b/net/ipv4/netfilter/nf_conntrack_l3proto_ipv4.c
index bbbfee7..10e970b 100644
--- a/net/ipv4/netfilter/nf_conntrack_l3proto_ipv4.c
+++ b/net/ipv4/netfilter/nf_conntrack_l3proto_ipv4.c
@@ -452,8 +452,6 @@ static void nf_ct_proto_ipv4_sysctl_cleanup(void)
 	if (!ve_is_super(get_exec_env())) {
 		free_sysctl_clone(ve_nf_conntrack_l3proto_ipv4->ctl_table);
 		ve_nf_conntrack_l3proto_ipv4->ctl_table = NULL;
-		kfree(ve_nf_conntrack_l3proto_ipv4);
-		ve_nf_conntrack_l3proto_ipv4 = NULL;
 	}
 }
 #else
-- 
1.5.4.3

